src/components/UserList/index.tsx

```diff
@@ -66,38 +160,93 @@ export default function UserList({ lang, users }: Props) {
         padding: '1rem',
       }}
     >
-      <Box sx={{ paddingBottom: '1rem' }}>
+      <Box sx={{ display: 'flex', paddingBottom: '1rem' }}>
         <Typography variant="h2">{t('EditUsers.label')}</Typography>
+        <Tooltip
+          title={
+            displayFilterRow
+              ? t('EditUsers.filterHideLabel')
+              : t('EditUsers.filterToggleLabel')
+          }
+        >
+          <IconButton
+            data-testId="filter-button"
```

Comment:
```suggestion
            data-testid="filter-button"
```
" Warning: React does not recognize the `data-testId` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `data-testid` instead. If you accidentally passed it from a parent component, remove it from the DOM element. "